hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -713,11 +714,24 @@ private void deleteInvalidFilesByPartitions(HoodieEngineContext context, Map<Str
           final HoodieStorage storage = metaClient.getStorage();
           LOG.info("Deleting invalid data file=" + partitionFilePair);
           // Delete
-          try {
-            storage.deleteFile(new StoragePath(partitionFilePair.getValue()));
-          } catch (IOException e) {
-            throw new HoodieIOException(e.getMessage(), e);
-          }
+          RetryHelper<Void, HoodieIOException> retryHelper = new RetryHelper(5000, 3, 5000, HoodieIOException.class.getName())
+              .tryWith(() -> {
```

Comment:
The retry helper actually starts a thread, there is concern we start too many threads in file deletion.

Reply:
I didn't find any place to start a thread.